egs/librispeech/asr/nnlm/run.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Xiaomi Corporation (Author: Liyong Guo)
+# Apache 2.0
+
+# References:
+# https://github.com/kaldi-asr/kaldi/blob/master/scripts/rnnlm/train_rnnlm.sh
+# https://github.com/kaldi-asr/kaldi/blob/master/scripts/rnnlm/prepare_rnnlm_dir.sh
+# https://github.com/pytorch/examples/tree/master/word_language_model
+# https://huggingface.co/docs/tokenizers/python/latest/quicktour.html
+
+# Example of how to use HuggingFace tokenizer and train {RNN, Transformer} based LMs
+
+set -e
+stage=$1
+
+lm_train=data/lm_train/
+full_text=$lm_train/librispeech_train_960_text
+tokenizer=$lm_train/tokenizer-librispeech_train_960.json
+if [ $stage -eq 1 ]; then
+  python3 ./local/download_lm_train_data.py
+fi
+if [ $stage -eq 2 ]; then
+  echo "training tokenizer"
+  python3 local/huggingface_tokenizer.py \
+    --train-file=$full_text \
+    --tokenizer-path=$tokenizer
+fi
+
+if [ $stage -eq 3 ]; then
+  echo "tokenize a file"
+  python3 local/huggingface_tokenizer.py \
+    --test-file=$full_text \
+    --tokenizer-path=$tokenizer
+fi
+
+if [ $stage -eq 4 ]; then
+  echo "split all data into train/valid/test"
+
+  full_tokens=${full_text}.tokens
+  valid_test_fraction=10 # currently 5 percent for valid and 5 percent for test
+  valid_test_tokens=$lm_train/valid_test.tokens
+  train_tokens=$lm_train/train.tokens
+
+  num_utts_total=$(wc -l <$full_tokens )
+  num_valid_test=$(($num_utts_total/${valid_test_fraction}))
+  set +x
+  shuf -n $num_valid_test  $full_tokens > $valid_test_tokens
```

Comment:
Shall we fix the seed for `shuf` so that the split is reproducible?
I think a Python script can do this task equally well and is more maintainable.

Reply:
Reproducible is important. Maybe the [data seperation method of kaldi RNNLM](https://github.com/kaldi-asr/kaldi/blob/pybind11/egs/librispeech/s5/local/rnnlm/tuning/run_tdnn_lstm_1a.sh#L75) can be used in following experiments.
` gunzip -c $text | cut -d ' ' -f2- | awk -v text_dir=$text_dir '{if(NR%2000 == 0) { print >text_dir"/dev.txt"; } else {print;}}' >$text_dir/librispeech.txt`

Reply:
+1 for dropping bash/perl entirely for these sorts of tasks in snowfall.